### PR TITLE
[android_alarm_manager] Android Code Inspection and Clean up

### PR DESCRIPTION
**Pull request by hamdikahloun**
## Description

- Handle deprecation & unchecked warning as error

- Avoiding uses or overrides a deprecated API in FlutterBackgroundExecutor.java

- 'while' loop replaced with enhanced 'for'

- 'if' statement replaced with 'switch' statement

- Remove Unnecessary 'Integer.toString()' call

A message is displayed when building for android: 

```
flutter build apk
Running "flutter pub get" in example...                             1,0s
You are building a fat APK that includes binaries for android-arm, android-arm64, android-x64.
If you are deploying the app to the Play Store, it's recommended to use app bundles or split the APK to reduce the APK
size.
    To generate an app bundle, run:
        flutter build appbundle --target-platform android-arm,android-arm64,android-x64
        Learn more on: https://developer.android.com/guide/app-bundle
    To split the APKs per ABI, run:
        flutter build apk --target-platform android-arm,android-arm64,android-x64 --split-per-abi
        Learn more on:  https://developer.android.com/studio/build/configure-apk-splits#configure-abi-split
Note: C:\Users\hamdi\Downloads\plugins-master\packages\android_alarm_manager\android\src\main\java\io\flutter\plugins\androidalarmmanager\FlutterBackgroundExecutor.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\Users\hamdi\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\espresso-0.0.1+5\android\src\main\java\androidx\test\espresso\flutter\EspressoFlutter.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Removed unused resources: Binary resource data reduced from 623KB to 581KB: Removed 6%
Running Gradle task 'assembleRelease'...
Running Gradle task 'assembleRelease'... Done                     104,3s
√ Built build\app\outputs\flutter-apk\app-release.apk (16.4MB).
```

```
flutter build appbundle
Running "flutter pub get" in example...                             1,1s
Note: C:\Users\hamdi\Downloads\plugins-master\packages\android_alarm_manager\android\src\main\java\io\flutter\plugins\androidalarmmanager\FlutterBackgroundExecutor.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\Users\hamdi\AppData\Local\Pub\Cache\hosted\pub.dartlang.org\espresso-0.0.1+5\android\src\main\java\androidx\test\espresso\flutter\EspressoFlutter.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Removed unused resources: Binary resource data reduced from 461KB to 405KB: Removed 12%
Running Gradle task 'bundleRelease'...
Running Gradle task 'bundleRelease'... Done                        78,5s
√ Built build\app\outputs\bundle\release\app.aab (16.5MB).
```

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/android_alarm_manager/android/gradle.properties [hunk 1]
```diff
@@ -1 +1,2 @@
 org.gradle.jvmargs=-Xmx1536M
+android.enableR8=true
```
Comment (dnfield) [inline packages/android_alarm_manager/android/gradle.properties:2]: I don't think we want this as part of this patch

File: packages/android_alarm_manager/android/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
```
Comment (dnfield) [inline packages/android_alarm_manager/android/gradle/wrapper/gradle-wrapper.properties:2]: Did you mean to add this file?

File: packages/android_alarm_manager/android/src/main/java/io/flutter/plugins/androidalarmmanager/AlarmService.java [hunk 3]
```diff
@@ -76,9 +75,9 @@ public static void startBackgroundIsolate(Context context, long callbackHandle)
     synchronized (alarmQueue) {
       // Handle all the alarm events received before the Dart isolate was
       // initialized, then clear the queue.
-      Iterator<Intent> i = alarmQueue.iterator();
-      while (i.hasNext()) {
-        flutterBackgroundExecutor.executeDartCallbackInBackgroundIsolate(i.next(), null);
+      /// 'while' loop replaceable with enhanced 'for'
```
Comment (dnfield) [inline packages/android_alarm_manager/android/src/main/java/io/flutter/plugins/androidalarmmanager/AlarmService.java:78]: Please remove thes e comments

Comment (hamdikahloun) [reply packages/android_alarm_manager/android/src/main/java/io/flutter/plugins/androidalarmmanager/AlarmService.java:78]: ah sorry my bad

File: packages/android_alarm_manager/android/gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (hamdikahloun) [reply packages/android_alarm_manager/android/gradle/wrapper/gradle-wrapper.properties:2]: no, i will fix it now and push again , Thank you @dnfield

File: packages/android_alarm_manager/android/gradle.properties (hunk 1 above)
Comment (hamdikahloun) [reply packages/android_alarm_manager/android/gradle.properties]: it's good, I took care of all the comments

File: packages/android_alarm_manager/android/gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (hamdikahloun) [reply packages/android_alarm_manager/android/gradle/wrapper/gradle-wrapper.properties]: it's good, I took care of all the comments

File: packages/android_alarm_manager/android/src/main/java/io/flutter/plugins/androidalarmmanager/AlarmService.java (hunk 3 above)
Comment (hamdikahloun) [reply packages/android_alarm_manager/android/src/main/java/io/flutter/plugins/androidalarmmanager/AlarmService.java]: i juste remove all e commants like 

> /// 'while' loop replaceable with enhanced 'for'

**Review by dnfield — APPROVED** [review]
LGTM, thanks!

**Pull request merged**